src/components/album/AlbumChat.js

```diff
@@ -1,23 +1,32 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import io from 'socket.io-client';
 
 const AlbumChat = ({ apiUrl }) => {
-  useEffect(() => {
+  const [message, setMessage] = useState('');
+
+  // Nandle sending a chat message
+  const handleSendMessage = () => {
     const socket = io(process.env.REACT_APP_SOCKET_BASE_PATH);
```

Comment:
Right now, this line is creating a new socket connection every time we send a message. You can see in the video below that each time I send a message, a brand new websocket creation is created and in the backend each message seems to be coming from a new socket id:

One way to fix this by saving the socket in a variable in component state (please note, I'm not sure 100% if this is the best practice in React-land, but it will work well enough for this use case :sweat\_smile: ). So a `[socket, setSocket] = useState()` call. And then when the component first mounts in your `useEffect` on line 20, connect with `io(...)` as you do already, and do a `setSocket(...)` to save that socket to the component state.

Then here in the `handleSendMessage` you can use that `socket` variable instead of reconnecting with `io(...)` again.

Reply:
<img width="730" alt="socket" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/95896159/1c9cfe64-4712-4e69-8df7-c517d14582e6">
this is what i have on backend now, when i sent message from /chat page input